job_dsl/rpc_gating_merge_trigger.groovy

```diff
@@ -16,58 +16,53 @@ common.globalWraps(){
     return
   }
 
-  stage("Build-Gating-Venv") {
-    build(
-      job: "Build-Gating-Venv",
-      wait: true,
-      parameters: [
-        [
-          $class: 'StringParameterValue',
-          name: 'RPC_GATING_BRANCH',
-          value: env.RPC_GATING_BRANCH
-        ]
-      ]
-    ) // build
-  } // stage
+  // Here we define the jobs to execute when a merge happens.
+  // Some are executed in series, and some parallel. We wait
+  // until they are all complete before completing the pipeline.
 
-  stage("Jenkins-Job-Builder") {
-    build(
-      job: "Jenkins-Job-Builder",
-      wait: true,
-      parameters: [
-        [
-          $class: 'StringParameterValue',
-          name: 'RPC_GATING_BRANCH',
-          value: env.RPC_GATING_BRANCH
-        ]
-      ]
-    ) // build
-  } // stage
+  List serialJobs = [
+    'Build-Gating-Venv',
```

Comment:
Jenkins-Job-Builder will start Build-Gating-Venv if the venv doesn't exist so this could be left out.

Reply:
Yes, that's true, but I'd prefer this to be explicit personally. Also, relying on the one job to kick off the other means further delays (wait for build, then wait a minute, then carry on), which isn't necessary.